src/ui/ui_util.c

```diff
@@ -128,11 +128,15 @@ static void _position_left(component_t* parent, component_t* child)
 
 static void _position_right(component_t* parent, component_t* child)
 {
-    child->position.left = parent->position.left + parent->dimension.width - child->dimension.width;
+    uint16_t left = 0;
```

Comment:
_position_right() without a parent does not make any sense, but if anything, it should start at SCREEN_WIDTH, otherwise it will not be right-positioned. 

Below `left - child->dimension.width` would also underflow.

Alternatively, Abort() if parent is NULL?